app/controllers/admin/base_controller.rb

```diff
@@ -9,11 +9,8 @@ def welcome
   private
 
   def redirect_or_edit(obj, saved, notice = nil)
-    if saved
-      redirect_to url_for_obj(obj), notice: notice_for(obj, notice)
-    else
-      render 'edit'
-    end
+    return redirect_to url_for_obj(obj), notice: notice_for(obj, notice) if saved
+    render 'edit'

```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.